### PR TITLE
test: refactor pagination snaps

### PR DESCRIPTION
**Pull request by craigbilner**
test: move to conform with new conventions and splits out extensive style snapshots
feat: allow `null` as a `replacer` for the `replace` serializer

### CODE REVIEW

File: packages/jest-serializer/__tests__/ios/replace.test.js [hunk 1]
```diff
@@ -35,6 +35,28 @@ describe("The replace serializer should", () => {
...
+  it("replace the expected component completely", () => {
+    addSerializers(
+      expect,
+      replace({
+        Text: null
+      })
+    );
+
+    // eslint-disable-next-line react/prop-types
+    const WrapperComponent = ({ children }) => <View>{children}</View>;
```
Comment (colincclark) [inline packages/jest-serializer/__tests__/ios/replace.test.js]: lots of repetition here, can we DRY it up?

File: packages/jest-serializer/__tests__/ios/__snapshots__/replace.test.js.snap [hunk 2]
```diff
@@ -55,3 +55,5 @@ exports[`The replace serializer should replace the expected component 1`] = `
   </Placeholder>
 </View>
 `;
+
+exports[`The replace serializer should replace the expected component completely 1`] = `<View />`;
```
Comment (colincclark) [inline packages/jest-serializer/__tests__/ios/__snapshots__/replace.test.js.snap]: can we lose the uppercase `describe` stuff ("The replace serializer should")

File: packages/pagination/__tests__/web/pagination.web.test.js [hunk 3]
```diff
@@ -1,5 +1,157 @@
...
+  minimalWebTransform,
+  replacePropTransform,
+  rnwTransform,
+  stylePrinter
+} from "@times-components/jest-serializer";
 import Pagination from "../../src/pagination";
 
-describe("Pagination on Web", tests(Pagination));
+describe("web", () => {
+  const hash = v =>
```
Comment (colincclark) [inline packages/pagination/__tests__/web/pagination.web.test.js]: this is a candidate for the upcoming `test-utils` package?

File: packages/jest-serializer/__tests__/web/__snapshots__/replace.test.js.snap [hunk 4]
```diff
@@ -85,3 +85,13 @@ exports[`The replace serializer should replace the expected component 1`] = `
   </View>
 </WrapperComponent>
 `;
+
+exports[`The replace serializer should replace the expected component completely 1`] = `
+<WrapperComponent>
+  <View>
+    <div
+      className="rn-alignItems-1oszu61 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1pxmb3b rn-flexBasis-7vfszb rn-flexDirection-eqz5dr rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-zIndex-1lgpqti"
```
Comment (colincclark) [inline packages/jest-serializer/__tests__/web/__snapshots__/replace.test.js.snap]: is this class name useful in a snapshot?

File: packages/jest-serializer/__tests__/ios/replace.test.js [hunk 5]
```diff
@@ -35,6 +35,28 @@ describe("The replace serializer should", () => {
...
+    // eslint-disable-next-line react/prop-types
+    const ChildComponent = ({ text }) => <Text>{text}</Text>;
+
+    const testInstance = TestRenderer.create(
+      <WrapperComponent>
+        <ChildComponent text="Hello world!" />
+      </WrapperComponent>
+    );
+
+    expect(testInstance.toJSON()).toMatchSnapshot();
```
Comment (colincclark) [inline packages/jest-serializer/__tests__/ios/replace.test.js]: these snapshots don't have names?

File: packages/pagination/__tests__/web/pagination.web.test.js (hunk 3 above)
Comment (craigbilner) [reply packages/pagination/__tests__/web/pagination.web.test.js]: think there's an obvious need for a `removeStyle` serializer too

File: packages/jest-serializer/__tests__/web/__snapshots__/replace.test.js.snap (hunk 4 above)
Comment (craigbilner) [reply packages/jest-serializer/__tests__/web/__snapshots__/replace.test.js.snap]: in two minds, with the serialization stuff tried to focus on the specific serializer under test

**Pull request merged**